server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,17 +370,55 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
+            // TODO: Add support for parsing the otel incoming tracer.
+            final SpanScope httpRequestSpanScope = telemetryTracer.startSpan(
```

Comment:
The `OpenTelemetry` has propagators that should be used to extract the context from request header, we may consider to introduce the `HttpTracer` interface (that `Tracer` would implement) to help us manage that:

```
protected TraceScopeHolder<TraceScope> startSpan(final Map<String, List<String>> headers, ....) {
        Context parent = openTelemetry.getPropagators().getTextMapPropagator()
            .extract(Context.current(), requestHeaders, ...);
        Scope parentScope = parent.makeCurrent();

        return startSpan(....); /* delegate to tracer here */
```

Reply:
Thanks for your input. Yes I am also thinking on the same lines. Will raise in the next PR.

Reply:
Tracking here - https://github.com/opensearch-project/OpenSearch/issues/9465